interface/model_init.py

```diff
@@ -0,0 +1,16 @@
+from models.linear.linear_model import linear_model
+from models.convolutional.cnn import cnn_model
+import torch
+from constants import LINEAR, CONVOLUTIONAL, DEVICE
+
+models = {
+    LINEAR: linear_model,
+    CONVOLUTIONAL: cnn_model,
+}
+
+
+def instantiate_model(architecture: str) -> torch.nn.Module:
+    nn_model = models[architecture]
+    model = nn_model.model().to(DEVICE)
```

Comment:
Do we instead want to let it choose? I've seen code which is along the lines of "cuda if cuda available else cpu"